tests/e2e/framework/kubernetes.go

```diff
@@ -237,6 +242,10 @@ func newKubeInfo(tmpDir, runID, baseVersion string) (*KubeInfo, error) {
 	}
 
 	a := NewAppManager(tmpDir, *namespace, i, kubeConfig)
+	var l *LocalRegistry
+	if os.Getenv("HUB") == "" {
```

Comment:
Could we instead, just go through all of the individual hub flags (i.e. mixerHub, pilotHub, caHub, proxyHub, galleyHub) and determine if any of them are set? I think this should be fairly straight forward ... all of the flags are defined in this file.